src/algebra/order/floor.lean

```diff
@@ -719,22 +753,72 @@ open int
 /-! ### Round -/
 
 section round
-variables [linear_ordered_field α] [floor_ring α]
+
+section linear_ordered_ring
+
+variables [linear_ordered_ring α] [floor_ring α]
 
 /-- `round` rounds a number to the nearest integer. `round (1 / 2) = 1` -/
-def round (x : α) : ℤ := ⌊x + 1 / 2⌋
+def round (x : α) : ℤ := if 2 * fract x < 1 then ⌊x⌋ else ⌈x⌉
+
+@[simp] lemma round_zero : round (0 : α) = 0 := by simp [round]
 
-@[simp] lemma round_zero : round (0 : α) = 0 := floor_eq_iff.2 (by norm_num)
-@[simp] lemma round_one : round (1 : α) = 1 := floor_eq_iff.2 (by norm_num)
+@[simp] lemma round_one : round (1 : α) = 1 := by simp [round]
```

Comment:
Can you also prove `round_nat_cast` and `round_int_cast`?

Reply:
Done.

The naming is slightly inconsistent since the corresponding lemmas for `floor` and `fract` are use `_coe` and `_coe'` suffixes. I'm not sure which we prefer.